### PR TITLE
docs(launch): focus job creation docs on `wandb launch` command

### PR DESCRIPTION
**Pull request by bcsherma**
## Description

What does the pull request do? If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Ticket

Does this PR fix an existing issue? If yes, provide a link to the ticket here:

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [ ] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [ ] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings.
- [ ] I merged the latest changes from `main` into my feature branch before submitting this PR.

### CODE REVIEW

**Comment by bcsherma** [comment]
* **#671** <a href="https://app.graphite.dev/github/pr/wandb/docodile/671?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bcsherma and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/guides/launch/create-launch-job.md [hunk 1]
```diff
@@ -10,234 +10,91 @@ import { CTAButtons } from '@site/src/components/CTAButtons/CTAButtons.tsx';
...
+4. Start the container and run `python job.py`.
 
-:::note
-You do not need to use the [`run.log_code()`](../../ref/python/run.md#log_code) function within your Python script when you create a launch job with the W&B CLI.
-:::
+To build from a branch or commit hash, append the `-g`, `--git-hash` argument. For a full list of arguments, run `wandb launch --help`.
 
-  </TabItem>
-  <TabItem value="sdk">
+### Remote urls
```
Comment (KyleGoyette) [inline docs/guides/launch/create-launch-job.md]: this is sentence style no? Whats up with this linter?

**Review by KyleGoyette — COMMENTED** [review]
The linters on this made it unreadable. I suspect some misbehavior from them considering the other comment I left.

We should still point users towards the create job utility function. Creating a job without running it immediately has some value for sweeps and automations.

File: docs/guides/launch/create-launch-job.md [hunk 2]
```diff
@@ -10,234 +10,97 @@ import { CTAButtons } from '@site/src/components/CTAButtons/CTAButtons.tsx';
...
-defaultValue="cli"
-values={[
-{label: 'CLI', value: 'cli'},
-{label: 'Python SDK', value: 'sdk'}
-]}>
-<TabItem value="cli">
-
-Create a launch job with the W&B CLI.
+:::tip
+To create a job without submitting it for exection, use the `wandb job create` command. See the [command reference docs](../../ref/cli/wandb-job/wandb-job-create.md) for more information.
```
Comment (KyleGoyette) [inline docs/guides/launch/create-launch-job.md]: execution*

File: docs/guides/launch/create-launch-job.md [hunk 3]
```diff
@@ -10,234 +10,97 @@ import { CTAButtons } from '@site/src/components/CTAButtons/CTAButtons.tsx';
...
-### Git remote URL handling
-
-The Git remote associated with a launch job can be either an HTTPS or an SSH URL. Git remote URLs typically use the following formats:
+## Automatic job creation
 
-- `https://github.com/organization/repository.git` (HTTPS)
-- `git@github.com:organization/repository.git` (SSH)
-
-The exact format varies by git hosting provider.
+W&B will automatically create and track a job for any run with tracked source code, even if that run was not created with Launch. Runs are considered to have tracked source code if any of the three following conditions are met:
```
Comment (KyleGoyette) [inline docs/guides/launch/create-launch-job.md:89]: Worth noting the job_source setting and the order of sources used here. IMO

**Review by KyleGoyette — COMMENTED** [review]
On first pass this seems great, but there's some important info dropped:
the job source setting, the order of sources checked by default, the WANDB_JOB_NAME env var. 

Are these things reflected elsewhere?

**Comment by bcsherma** [comment]
> On first pass this seems great, but there's some important info dropped: the job source setting, the order of sources checked by default, the WANDB_JOB_NAME env var.
> 
> Are these things reflected elsewhere?

The `WANDB_JOB_NAME` and the `job_name` setting are still in here. What is the `job_source` setting? That doesn't appear anywhere in our docs

File: docs/guides/launch/create-launch-job.md [hunk 4]
```diff
@@ -10,234 +10,97 @@ import { CTAButtons } from '@site/src/components/CTAButtons/CTAButtons.tsx';
...
-Create a launch job with the W&B CLI.
+:::tip
+To create a job without submitting it for execution, use the `wandb job create` command. See the [command reference docs](../../ref/cli/wandb-job/wandb-job-create.md) for more information.
+:::
 
-Ensure the path with your Python script has a `requirements.txt` file with the Python dependencies required to run your code. A Python runtime is also required. The python runtime can either be specified manually with the runtime parameter or can be auto-detected from a `runtime.txt` or `.python-version file`.
+## Git jobs
 
-Copy and paste the following code snippet. Replace the values within `"<>"` based on your use case:
+Create a Git-based job with W&B Launch. Code and other assets are cloned from a certain commit, branch, or tag in a git repository. Run a "hello world" job from a git repository with the following command:
```
Comment (KyleGoyette) [inline docs/guides/launch/create-launch-job.md]: "Create a Git-based job with W&B Launch" without a short description of what jobs are feels weird. Lets add back some info about Jobs capturing the source code and environment

File: docs/guides/launch/create-launch-job.md [hunk 5]
```diff
@@ -10,234 +10,97 @@ import { CTAButtons } from '@site/src/components/CTAButtons/CTAButtons.tsx';
...
 
 ```bash
-wandb job create --project "<project-name>" -e "<your-entity>" \
---name "<name-for-job>" code "<path-to-script/code.py>"
+wandb launch --uri "https://github.com/wandb/launch-jobs.git" --build-context jobs/hello_world --dockerfile Dockerfile.wandb --project "hello-world" --job-name "hello-world" --entry-point "python job.py"
 ```
 
-For a full list of flags you can use, see the [`wandb job create`](../../ref/cli/wandb-job/wandb-job-create.md) command documentation.
+The command does the following:
+1. Clone the [W&B Launch jobs repository](https://github.com/wandb/launch-jobs) to a temporary directory.
```
Comment (KyleGoyette) [inline docs/guides/launch/create-launch-job.md]: All of these verbs should be third person singular form in the present tense. Ie put an s on "Clones", "Creates" "Builds" and "Starts and runs"

File: docs/guides/launch/create-launch-job.md [hunk 6]
```diff
@@ -10,234 +10,97 @@ import { CTAButtons } from '@site/src/components/CTAButtons/CTAButtons.tsx';
...
-        entity="<your-entity>",
-        # Simulate tracking hyperparameters
-        config={
-            "learning_rate": lr,
-            "epochs": epochs,
-        },
-    )
+| Remote URL | Requirements for access and authentication |
+| ---------- | ------------------------------------------ |
+| https URL  | username and password to authenticate with the git remote |
```
Comment (KyleGoyette) [inline docs/guides/launch/create-launch-job.md]: Can we drop "URL" in these? And it may make sense to combine the examples above in this table. So its:

Remote URL | URL format | requirements for access and authentication

File: docs/guides/launch/create-launch-job.md [hunk 7]
```diff
@@ -10,234 +10,97 @@ import { CTAButtons } from '@site/src/components/CTAButtons/CTAButtons.tsx';
...
+| ssh URL    | ssh key to authenticate with the git remote |
 
-    offset = random.random() / 5
-    print(f"lr: {lr}")
+The Git remote URL is inferred from the local git repository if your launch job is created automatically by a W&B run.
```
Comment (KyleGoyette) [inline docs/guides/launch/create-launch-job.md]: Move this into automatic job creation

File: docs/guides/launch/create-launch-job.md [hunk 8]
```diff
@@ -10,234 +10,97 @@ import { CTAButtons } from '@site/src/components/CTAButtons/CTAButtons.tsx';
...
 
-    # highlight-next-line
-    run.log_code()
-    run.finish()
 
+## Code artifact jobs
 
-run_training_run(epochs=10, lr=0.01)
-```
+Jobs can be created from any source code stored in a W&B Artifact. Pass a local code directory to the `--uri` argument to create a new code artifact and job.
```
Comment (KyleGoyette) [inline docs/guides/launch/create-launch-job.md]: "Pass a local code directory..." -> "Use a local directory..."

File: docs/guides/launch/create-launch-job.md [hunk 9]
```diff
@@ -10,234 +10,97 @@ import { CTAButtons } from '@site/src/components/CTAButtons/CTAButtons.tsx';
...
-│   job.py
-│   requirements.txt
-└───configs
-│   │   example.yml
+with wandb.init() as run:
+    run.log({"metric": 0.5})
 ```
 
-You can create a Docker image called `fashion-mnist` with the `docker build` command:
+Add a file `requirements.txt` with the following content:
```
Comment (KyleGoyette) [inline docs/guides/launch/create-launch-job.md:60]: We should maybe indicate in a callout this prerequisite.

**Review by KyleGoyette — COMMENTED** [review]
I think we should also add a bit at the bottom about how jobs can be enqueued by adding the `q` arg to the CLI command

File: docs/guides/launch/create-launch-job.md [hunk 10]
```diff
@@ -10,234 +10,100 @@ import { CTAButtons } from '@site/src/components/CTAButtons/CTAButtons.tsx';
...
-Create a launch job with the W&B CLI.
+:::tip
+To create a job without submitting it for execution, use the `wandb job create` command. See the [command reference docs](../../ref/cli/wandb-job/wandb-job-create.md) for more information.
+:::
 
-Ensure the path with your Python script has a `requirements.txt` file with the Python dependencies required to run your code. A Python runtime is also required. The python runtime can either be specified manually with the runtime parameter or can be auto-detected from a `runtime.txt` or `.python-version file`.
+## Git jobs
 
-Copy and paste the following code snippet. Replace the values within `"<>"` based on your use case:
+Create a Git-based job with W&B Launch. Code and other assets are tracked and cloned from a certain commit, branch, or tag in a remote git repository. Run a "hello world" job from a git repository with the following command:
```
Comment (KyleGoyette) [inline docs/guides/launch/create-launch-job.md]: "Create a Git-based job where code and other tracked assets are cloned from a certain commit, branch or tag in a remote git repository with W&B Launch. Run a "hello world" job from a git repository with the following command:"

What do you think of that?

File: docs/guides/launch/create-launch-job.md [hunk 11]
```diff
@@ -10,234 +10,100 @@ import { CTAButtons } from '@site/src/components/CTAButtons/CTAButtons.tsx';
...
-        },
-    )
+The remote url format determines how the git remote is accessed and authenticated:
 
-    offset = random.random() / 5
-    print(f"lr: {lr}")
+| Remote URL | Requirements for access and authentication |
+| ---------- | ------------------------------------------ |
+| https      | username and password to authenticate with the git remote |
+| ssh        | ssh key to authenticate with the git remote |
```
Comment (KyleGoyette) [inline docs/guides/launch/create-launch-job.md]: I still think these should all be in one table:
| Remote URL Type| URL Format | Requirements for access and authentication |
| ----------| ------------------- | ------------------------------------------ |
| https      | `https://github.com/organization/repository.git`  | username and password to authenticate with the git remote |
| ssh        | `git@github.com:organization/repository.git` | ssh key to authenticate with the git remote |

File: docs/guides/launch/create-launch-job.md [hunk 12]
```diff
@@ -280,3 +144,29 @@ wandb.init(settings=settings)
...
+Jobs are executed in a container. Image jobs use a pre-built Docker image, while Git and code artifact jobs require a container build step.
+
+Job containerization can be customized with arguments to `wandb launch` and files within the job source code.
+
+### Build context
+
+The term build context refers to the tree of files and directories that are sent to the Docker daemon to build a container image. By default, Launch uses the root of the job source code as the build context. To specify a subdirectory as the build context, use the `--build-context` argument of `wandb launch` when creating and launching a job.
+
+:::tip
+The `--build-context` argument is particularly useful for working with Git jobs that refer to a monorepo with multiple projects. By specifying a subdirectory as the build context, you can build a container image for a specific project within the monorepo.
```
Comment (KyleGoyette) [inline docs/guides/launch/create-launch-job.md:152]: Isnt the dockerfile arg useful here as well? Especially if we want to remove automatic dockerfile finding?

**Review by KyleGoyette — COMMENTED** [review]
Why not put the git url stuff into one table? The addition of containerization and requirements file are great!

File: docs/guides/launch/create-launch-job.md (hunk 12 above)
Comment (bcsherma) [reply docs/guides/launch/create-launch-job.md:152]: I separated the documentation of build context and dockerfile customization because build context is also relevant for requirements files. I think this arg is specifically useful for git jobs because there is really no reason for a code job to have a build context other than the root of the source code, it's only really useful for monorepos

File: docs/guides/launch/create-launch-job.md [hunk 13]
```diff
@@ -10,234 +10,93 @@ import { CTAButtons } from '@site/src/components/CTAButtons/CTAButtons.tsx';
...
-wandb job create --project "<project-name>" --entity "<your-entity>" \
---name "<name-for-job>" image image-name:tag
+wandb launch --uri . --job-name hello-world-code --project launch-quickstart --entry-point "python train.py"
 ```
 
-For a full list of flags you can use, see the [`wandb job create`](../../ref/cli/wandb-job/wandb-job-create.md) command documentation.
+The preceding command does the following:
+1. Logs the current directory as a code artifact named `hello-world-code`.
+2. Creates a job named `hello-world-code` in the `launch-quickstart` project.
+3. Builds a container image from the current directory and Launch's default Dockerfile. The default dockerfile will install the `requirements.txt` file and set the entry point to `python train.py`.
```
Comment (KyleGoyette) [inline docs/guides/launch/create-launch-job.md]: dockerfile -> Dockerfile

**Review by KyleGoyette — APPROVED** [review]
Approved pending the fix on the format on the sidebar and the spelling of "dockerfile" in this PR

File: docs/guides/launch/create-launch-job.md [hunk 14]
```diff
@@ -10,234 +10,93 @@ import { CTAButtons } from '@site/src/components/CTAButtons/CTAButtons.tsx';
...
+1. Logs the current directory as a code artifact named `hello-world-code`.
+2. Creates a job named `hello-world-code` in the `launch-quickstart` project.
+3. Builds a container image from the current directory and Launch's default Dockerfile. The default Dockerfile will install the `requirements.txt` file and set the entry point to `python main.py`.
 
-  </TabItem>
-  <TabItem value="build">
+## Image jobs
 
-Associate your run with a Docker image. W&B looks for an image tag in the `WANDB_DOCKER` environment variable, and if `WANDB_DOCKER` is set, a launch job is created from the specified image tag. Ensure that the `WANDB_DOCKER` environment variable is set to the full image tag.
+Pre-made images can be used to create jobs. The image is pulled from a Docker registry and run with the specified entry point, or the default entry point if none is specified. Pass a full image tag to the `--docker-image` option to create and run a job from a Docker image.
```
Comment (jcrawley) [inline docs/guides/launch/create-launch-job.md:79]: I do think changing passive voice here is a legitimate suggestion.

**Review by jcrawley — APPROVED** [review]
